rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/example/boot/security/BootWebSecurityConfigurer.java

```diff
@@ -123,15 +125,22 @@ protected void addLoginPageConfiguration(HttpSecurity http) throws Exception {
             authorizeHttpRequests -> authorizeHttpRequests.anyRequest().fullyAuthenticated())
         .formLogin(
             formLogin ->
-                formLogin.loginPage("/login").failureUrl("/login?error").defaultSuccessUrl("/"))
+                formLogin
+                    .loginPage("/login")
+                    .failureUrl("/login?error")
+                    .defaultSuccessUrl("/index.html")
+                    .permitAll()
```

Comment:
Sure that `permitAll` is correct here? I am no spring security expert but this sounds weird.

Reply:
isn't this just for the login page path?

Reply:
Hello @mustaphazorgati , @sfrevel ,

@sfrevel is absolutely right.

The `permitAll()` specified here only refers to the authentication part. It is a bit confusing that you specify the `defaultSuccessUrl("/index.html")` and you might think that this is also permitted. However, the AuthenticationHandler ensures that this is not the case.

Without the `permitAll()` there is an infinite redirect to the default page `/taskana/index.html` -> which is secured and tries to call the login page. 

Only the login url and the login failure url are allowed here. Nothing else.

See the Spring security code for more details. I did not bother to look for the explanation in the Spring Security docs. 
* `org.springframework.security.config.annotation.web.configurers.FormLoginConfigurer`
* `org.springframework.security.config.annotation.web.configurers.AbstractAuthenticationFilterConfigurer`
* `org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter`
* `org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter`

Alternatively, we could specify the login url and the login failure url as exceptions above. (where the exceptions for CSS and images are defined). **Should I try this?**

I have tested the current config as follows

- user is not logged in
- call `localhost:8080/taskana` -> redirect to login
- do not log in!
- call `localhost:8080/taskana/index.html` -> redirect to login page
- call `localhost:8080/taskana/taskana/something` -> redirect to login page
- call `localhost:8080/taskana/#/taskana/administration/workbaskets` -> redirect to login page

login user now -> everything works as expected

is this enough explanation? or should we have a look at the code together?

Reply:
This is fine for me. Thank you for the explanation!